### PR TITLE
Added support for Provider EKS Cluster

### PR DESCRIPTION
**Pull request by asutoshpalai**
1. **What this PR does / why we need it**:

This PR adds support for the adoption of existing EKS clusters onto TMC via Provider EKS Cluster resource.

2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

3. **Additional information**

4. **Special notes for your reviewer**

### CODE REVIEW

**Comment by asutoshpalai** [comment]
As the acceptance test fails on the main branch, I've attached only the relevant successful test for the resources affected by this PR.

image

File: internal/resources/provider_ekscluster/data_source.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		diags diag.Diagnostics
+		resp  *models.VmwareTanzuManageV1alpha1ManageEksProvidereksclusterGetProviderEksClusterResponse
+		err   error
+	)
+
+	clusterFn := constructFullname(d)
+	getEksClusterResourceRetryableFn := func() (retry bool, err error) {
+		resp, err = config.TMCConnection.ProviderEKSClusterResourceService.ProviderEksClusterResourceServiceGet(clusterFn)
+		if err != nil {
+			if clienterrors.IsNotFoundError(err) {
```
Comment (shreyassreenivas) [inline internal/resources/provider_ekscluster/data_source.go:45]: In the case of data source read, we should return error here

File: resource_templates/provider_ekscluster.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  ready_wait_timeout = "<time>" // Wait time for cluster operations to finish (default: 30m).
+
+  meta {
+    description = "description of the cluster"
+    labels      = { "<key>" : "<value>" }
+  }
+
+  spec {
+    cluster_group = "test-cluster-group" // Default: default
```
Comment (shreyassreenivas) [inline resource_templates/provider_ekscluster.tf:25]: Have you tested a scenario where you do not provide cluster_gourp and in subsequent apply , there is no diff reported ?

**Comment by shreyassreenivas** [comment]
Also, can you rename the resource as per the discussion with PM

**Comment by asutoshpalai** [comment]
I am closing this PR while we are reevaluating the workflow for managing existing clusters via the Terraform Provider.

**Pull request closed without merging**